### PR TITLE
Snapshot testing for Introspection Engine + PSL

### PR DESCRIPTION
**Pull request by pimeys**
First small step making introspection engine using snapshot tests. Focus on constraint names and referential actions at this point to reveal possible problems there if we break something.

### CODE REVIEW

File: introspection-engine/introspection-engine-tests/src/test_api.rs [hunk 1]
```diff
@@ -81,12 +81,18 @@ impl TestApi {
...
         let introspection_result = self.test_introspect_internal(Datamodel::new()).await?;
+
         Ok(datamodel::render_datamodel_and_config_to_string(
             &introspection_result.data_model,
             &self.configuration(),
         ))
     }
 
+    pub async fn introspect_dml(&self) -> Result<String> {
+        let introspection_result = self.test_introspect_internal(Datamodel::new()).await?;
```
Comment (do4gr) [inline introspection-engine/introspection-engine-tests/src/test_api.rs:92]: I still don't like that we're passing in an empty datamodel here. That would not work in the rpc call. I'd love to have the generator and datasource in here to be sure it does the same as the CLI calls.

Comment (pimeys) [reply introspection-engine/introspection-engine-tests/src/test_api.rs:92]: It's another pass we should do later on. If we call `SqlIntrospectionConnector` directly we don't really do anything with the `DataModel`.

File: libs/datamodel/core/src/lib.rs [hunk 2]
```diff
@@ -201,9 +201,9 @@ fn load_sources(
 //
 
 /// Renders to a return string.
-pub fn render_datamodel_to_string(datamodel: &dml::Datamodel) -> String {
+pub fn render_datamodel_to_string(datamodel: &dml::Datamodel, configuration: Option<&Configuration>) -> String {
     let mut writable_string = String::with_capacity(datamodel.models.len() * 20);
-    render_datamodel_to(&mut writable_string, datamodel, None);
+    render_datamodel_to(&mut writable_string, datamodel, configuration);
```
Comment (tomhoule) [inline libs/datamodel/core/src/lib.rs:206]: Is this what the IE was using? I'm finding the rendering codepath hard to picture.

Comment (pimeys) [reply libs/datamodel/core/src/lib.rs:206]: Yeah, this is kind of confusing... So we have the function that renders config and dml that's using the config specialties for rendering and then we have the one that renders only the dml that didn't get the config at all. Now we kind of need to know the connector and preview features in this point too...

Comment (tomhoule) [reply libs/datamodel/core/src/lib.rs:206]: Ok yes now with a bit more time and context I think I understand.

**Review by tomhoule — COMMENTED** [review]
Only the tests where mysql/mssql return different datamodels left, I'll finish reviewing tomorrow

**Review by tomhoule — APPROVED** [review]

**Comment by tomhoule** [comment]
btw I think we should manifest differences between connectors as different modules rather than in the test name, eventually

**Comment by pimeys** [comment]
I'll do that from now on.

File: libs/datamodel/core/src/ast/expression.rs [hunk 3]
```diff
@@ -66,7 +85,7 @@ impl Expression {
         }
     }
 
-    /// Creates a friendly readable representation for a value's type.
+    /// Creates a friendly readable representation for a value"s type.
```
Comment (tomhoule) [inline libs/datamodel/core/src/ast/expression.rs:88]: ?

**Pull request merged**